From 1fd5b850a4260211b90831b13d847567081f3ce7 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Fri, 18 May 2018 16:00:25 +0200
Subject: [PATCH] Add ARM support into xf86arm.h. This provides support for
 Xorg interface. Without this the vivante samples will hang during close
 requiring a reboot

Upstream-Status: Pending

Signed-off-by: Lauren Post <lauren.post@freescale.com>
Signed-off-by: Evan Kotara <evan.kotara@freescale.com>
---
 xf86drm.h | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/xf86drm.h b/xf86drm.h
index 7773d71a..00a1ec42 100644
--- a/xf86drm.h
+++ b/xf86drm.h
@@ -469,6 +469,23 @@ do {	register unsigned int __old __asm("o0");		\
 		: "cr0", "memory");			\
 	} while (0)
 
+#elif defined(__arm__)
+       #undef DRM_DEV_MODE
+       #define DRM_DEV_MODE     (S_IRUSR|S_IWUSR|S_IRGRP|S_IWGRP|S_IROTH|S_IWOTH)
+
+       #define DRM_CAS(lock,old,new,__ret)             \
+       do {                                            \
+               __asm__ __volatile__ (                  \
+                       "1: ldrex %0, [%1]\n"           \
+                       "   teq %0, %2\n"               \
+                       "   ite eq\n"                   \
+                       "   strexeq %0, %3, [%1]\n"     \
+                       "   movne   %0, #1\n"           \
+               : "=&r" (__ret)                         \
+               : "r" (lock), "r" (old), "r" (new)      \
+               : "cc","memory");                       \
+       } while (0)
+
 #endif /* architecture */
 #endif /* __GNUC__ >= 2 */
 
-- 
2.23.0

